src/uint/modular/runtime_mod.rs

```diff
@@ -37,8 +37,18 @@ pub struct DynResidueParams<const LIMBS: usize> {
 }
 
 impl<const LIMBS: usize> DynResidueParams<LIMBS> {
-    /// Instantiates a new set of `ResidueParams` representing the given `modulus`.
+    #[deprecated(
+        since = "0.5.3",
+        note = "This will return an `Option` in a future version to account for an invalid modulus, but for now will panic if this happens. Consider using `new_checked` until then."
+    )]
+    /// Instantiates a new set of `ResidueParams` representing the given `modulus`, which _must_ be odd.
+    /// If `modulus` is not odd, this function will panic; use [`new_checked`] if you want to be able to detect an invalid modulus.
     pub const fn new(modulus: &Uint<LIMBS>) -> Self {
+        // The modulus must be odd
+        if modulus.ct_is_odd().to_u8() != 1 {
+            panic!("Modulus must be odd.");
+        }
+
```

Comment:
You should be able to do something like:

```suggestion
        match Self::new_checked(modulus) {
            Some(ret) => ret,
            None => panic!("modulus must be odd")
        }
```

Reply:
Done, thanks